src/nimble.nim

```diff
@@ -400,76 +400,81 @@ proc installFromDir(dir: string, requestedVer: VersionRange, options: Options,
                   @[]
     buildFromDir(pkgInfo, paths, "-d:release" & flags, options)
 
-  let pkgDestDir = pkgInfo.getPkgDest(options)
-  if dirExists(pkgDestDir) and fileExists(pkgDestDir / "nimblemeta.json"):
-    let msg = "$1@$2 already exists. Overwrite?" %
-              [pkgInfo.name, pkgInfo.specialVersion]
-    if not options.prompt(msg):
-      raise NimbleQuit(msg: "")
-
-    # Remove reverse deps.
-    let pkgInfo = getPkgInfo(pkgDestDir, options)
-    options.nimbleData.removeRevDep(pkgInfo)
-
-    removePkgDir(pkgDestDir, options)
-    # Remove any symlinked binaries
-    for bin in pkgInfo.bin:
-      # TODO: Check that this binary belongs to the package being installed.
-      when defined(windows):
-        removeFile(binDir / bin.changeFileExt("cmd"))
-        removeFile(binDir / bin.changeFileExt(""))
-      else:
-        removeFile(binDir / bin)
-
-  createDir(pkgDestDir)
-  # Copy this package's files based on the preferences specified in PkgInfo.
-  var filesInstalled = initHashSet[string]()
-  iterInstallFiles(realDir, pkgInfo, options,
-    proc (file: string) =
-      createDir(changeRoot(realDir, pkgDestDir, file.splitFile.dir))
-      let dest = changeRoot(realDir, pkgDestDir, file)
-      filesInstalled.incl copyFileD(file, dest)
-  )
-
-  # Copy the .nimble file.
-  let dest = changeRoot(pkgInfo.myPath.splitFile.dir, pkgDestDir,
-                        pkgInfo.myPath)
-  filesInstalled.incl copyFileD(pkgInfo.myPath, dest)
+  # Don't copy artifacts if local deps mode and "installing" the top level package
+  if not (options.localdeps and options.startDir == dir):
```

Comment:
Suggestion: create a ``isInstallingTopLevel`` function (or similar name) which is implemented via ``options.startDir == dir``. Seems like it will be useful for the future (and will make the explanation tied to the code).

Reply:
You're actually using the same code below, so it would already be used across two call sites, so I think that supports my suggestion.

Reply:
This is done.